src/python/pants/backend/project_info/dependencies_test.py

```diff
@@ -209,3 +215,165 @@ def test_python_dependencies(rule_runner: PythonRuleRunner) -> None:
         ],
         closed=True,
     )
+
+
+def test_python_dependencies_output_format_json_direct_deps(rule_runner: PythonRuleRunner) -> None:
```

Comment:
I wonder if it would be worth making a function of "expected with DependenciesOutputFormat.json"->"expected with DependenciesOutputFormat.merge" which we would invoke inside of assert_deps. This would allow us to pass the same value for `expected`, so we could parametrise the tests instead of duplicating by output format.

Reply:
Thanks for the suggestion. I agree that to parametrize tests is a great idea and often results in a bit less code and somewhat more concise. If the inputs/outputs would have been smaller, I'd definitely done that, but with lots large JSON pieces, I just feel that I lose the order of arguments and it actually takes me longer to reason about the test cases. So I felt like a bit of duplication of the inputs for each individual test case handled separate is justified to make it clear which use case is being tested. Am I too old-fashioned? :)

Reply:
I like the explicitness in this case. Perhaps the setup could be placed in a fixture though so that part could be re-used. No biggie either, tho.